bin/diffenv

```diff
@@ -1,9 +1,162 @@
-#!/usr/bin/env bash
-
-# suppress colors if we are not outputting to terminal
-if [ -t 1 ]
-then
-    diffenv-nopaging "$@" | less -FRX
-else
-    diffenv-nopaging --no-color "$@" | less -FRX
-fi
+#!/usr/bin/env python3
+
+import sys
+import argparse
+import os
+from shutil import copyfile, copystat
+import requests
+from io import StringIO
+from signal import signal, SIGPIPE, SIG_DFL
+import subprocess
+
+from diffenv import main, diffviewer, editor
+
+# Handle arguments
+parser = argparse.ArgumentParser(
+    description='Diff your total environment. Run without any params to simply output current environment state as YAML.',
+    epilog='More information and source code at https://github.com/error-central/diffenv')
+parser.add_argument('-o', '--output', help='output to file instead of stdout')
+parser.add_argument(
+    '-c',
+    '--compare',
+    help='file or URL to compare against the current env')
+parser.add_argument(
+    '--add-hooks',
+    action='store_true',
+    help='install git hooks in current repo and exit')
+parser.add_argument(
+    '--share',
+    action='store_true',
+    help='store current env and return URL that can be shared')
+parser.add_argument(
+    '--post',
+    default='https://transfer.sh',
+    help='POST env to specific URL when sharing; must be used with --share')
+parser.add_argument('--config', help='load config from specific file')
+parser.add_argument('--ignore-config',
+                    action='store_true',
+                    help='ignore configs and run all facets')
+parser.add_argument('--no-color',
+                    action='store_true',
+                    help="Don't color diff output")
+parser.add_argument('--no-paging',
+                    action='store_true',
+                    help="Don't use less for paging output")
+
+args = parser.parse_args()
+if args.output or args.share:
+    args.no_color = True
+
+if not sys.stdout.isatty():
+    # We're being piped or redirected, don't page
+    args.no_paging = True
+
+
+# Handle outputting to file, pager or stdout
+try:
+    if args.output:
+        outfilestream = open(args.output, 'w')
+    elif args.no_paging:
+        outfilestream = sys.stdout
+    else:
+        # args stolen fron git source, see `man less`
+        pager = subprocess.Popen(['less', '-F', '-R', '-S', '-X', '-K'],
+                                 stdin=subprocess.PIPE,
+                                 stdout=sys.stdout)
+
+        outfilestream = pager.stdin
+
+
+    signal(SIGPIPE, SIG_DFL)
```

Comment:
What does this line do? Seems like a magical incantation!

Reply:
I think it's just a relic from when we were using less -- sometimes the unix pipes would get messed up and it would crash -- we can remove it for now and see if we don't need it.

Reply:
Actually, we probably should leave it in -- since if somebody were to use less or some other pipe processing tool that takes long, but then doesn't consume the whole stream contents, they'd have the same problems, so I'll just add a comment